flink-libraries/flink-sql-client/src/main/java/org/apache/flink/table/client/config/Environment.java

```diff
@@ -91,6 +92,22 @@ public Deployment getDeployment() {
 		return deployment;
 	}
 
+	public String explain() {
+		final StringBuilder sb = new StringBuilder();
+		sb.append("===================== Tables =====================\n");
+		tables.forEach((name, table) -> {
+			sb.append("- name: ").append(name).append("\n");
+			final DescriptorProperties props = new DescriptorProperties(true);
+			table.addProperties(props);
```

Comment:
Should we make a copy of `table` before modifying it?

Reply:
The properties are added to `props` not `table`. Maybe the name is a bit misleading.